### PR TITLE
Add HDriveModel

### PR DESCRIPTION
**Pull request by Octogonapus**
### Description of the Change

This PR adds support for h-drives via `HDriveModel`.

### Motivation

A user was asking for this recently.

### Possible Drawbacks

None.

### Verification Process

Unit tests only. I have not verified this on real hardware.

### Applicable Issues

None.

### CODE REVIEW

**Comment by Octogonapus** [comment]
@theol0403 Could you please take a look over this?

**Comment by Octogonapus** [comment]
Anyone looking to test this, the artifact is here: [template.zip](https://github.com/OkapiLib/OkapiLib/files/3796370/template.zip)

**Comment by Sleuth56** [comment]
I've tested it. It works fine. It might be good to add a third controller input to the arcade drive. I got around it with a separate line controlling the middle motor.
`    driveTrain->arcade(controller.getAnalog(ControllerAnalog::leftY),controller.getAnalog(ControllerAnalog::rightX), 0.3);
    driveTrain->middle(controller.getAnalog(ControllerAnalog::leftX));`

**Comment by Octogonapus** [comment]
> It might be good to add a third controller input to the arcade drive

Use `hArcade` for that. `arcade` is an overridden method.

**Comment by theol0403** [comment]
@lerker100 
I'm guessing that you don't know how to use `hArcade` from a `ChassisController`.
The answer is that you need to get the model from the chassis (which is internally an `HDriveModel`, but gets returned as a generic `ChassisModel`), and then cast (tell the compiler what *type* of model you have) the `ChassisModel` into a `HDriveModel`.
For an example doing the same thing with an x-drive, check this out:
https://www.vexforum.com/t/xdriver-model-does-not-have-xarcade/67778/4?u=theol0403

File: src/api/chassis/model/hDriveModel.cpp [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  } else {
+    maxVelocity = imaxVelocity;
+  }
+}
+
+double HDriveModel::getMaxVelocity() const {
+  return maxVelocity;
+}
+
+void HDriveModel::setMaxVoltage(double imaxVoltage) {
```
Comment (theol0403) [inline src/api/chassis/model/hDriveModel.cpp:229]: I'm not sure if this is the same thing for all the models, but this function can be simplified to
```cpp
maxVoltage = std::clamp(imaxVoltage, 0, 12000);

**Review by theol0403 — CHANGES_REQUESTED** [review]
In functions that do not use the middle wheel, I recommend stopping the middle wheel. 
That way, doing 
```cpp
hArcade(1, 1, 0);
delay(200);
rotate(1);
delay(500);
```
won't make the robot try to strafe while rotating.

The rest looks good to me, though I did not fully go through the builder logic.

**Review by theol0403 — APPROVED** [review]

**Comment by Sleuth56** [comment]
Yea I found it after I said that but wasn't able to test it on hardware yet

**Pull request merged**